src/util/threadpool.rs

```diff
@@ -569,4 +665,112 @@ mod test {
         assert_eq!(task.gid, group2);
         queue.on_task_started(&group2);
     }
+
+
+    #[test]
+    fn test_small_group_first_queue() {
+        let concurrency_limit = 2;
+        let small_group_tasks_limit = 2;
+        let mut queue = SmallGroupFirstQueue::new(concurrency_limit, small_group_tasks_limit);
+        let mut id = 1;
+        let mut pre_id = 0;
+
+        // high_priority_queue and low_priority_queue is both empty.
+        assert!(queue.pop().is_none());
+
+        // Push 2 tasks of `group1` into queue
+        let group1 = 1001;
+        for _ in 0..small_group_tasks_limit {
+            let mut task = Task::new(group1, move || {});
+            task.id = id;
+            id += 1;
+            queue.push(task);
+        }
+
+        // high:[g11,g12]; low:[]
+        assert!(queue.low_priority_queue.is_empty());
+
+        // low_priority_queue is empty
+        let task = queue.pop().unwrap();
+        assert!(task.id > pre_id);
+        pre_id = task.id;
```

Comment:
Rename `pre_id` to `expected_task_id`, initialize it to id 1, and use `assert_eq!(task.id, expected_task_id)` anywhere `pre_id` is used. 
The point to do this change is to make the code simple, and make the assertion as fixed as possible.